### ISSUE TITLE
Not getting expected towncrier version

### ISSUE THREAD

**Issue by altendky** [twisted/towncrier#304]
Both master and your branch from #303 are still at 19.2.0, I only found https://github.com/twisted/towncrier/tree/hawkowl/release-19.09 which seems to be newer, but there I get the same:
```
% towncrier build --draft
Usage: towncrier build [OPTIONS]

Error: Got unexpected extra argument (build)
```

_Originally posted by @fschulze in https://github.com/twisted/towncrier/issues/105#issuecomment-746106228_

**Comment by altendky** [comment]
I'm more curious how you are installing things.  Also, `type towncrier`. Present `master` definitely seems to be coded for subcommands and I haven't run into any oddities in this regard.  It seems that for some reason you aren't running what you think you are.

https://github.com/twisted/towncrier/blob/ef1be451c44b71112cf6b6b2f1e90a27dd5ff489/setup.py#L31
https://github.com/twisted/towncrier/blob/ef1be451c44b71112cf6b6b2f1e90a27dd5ff489/src/towncrier/_shell.py#L16-L23

**Comment by fschulze** [comment]
The installation method seems to be the key.

With pip it works properly and the towncrier script uses after setting up the requirements for ``load_entry_point``:
```
...
if __name__ == '__main__':
    sys.argv[0] = re.sub(r'(-script\.pyw?|\.exe)?$', '', sys.argv[0])
    sys.exit(load_entry_point('towncrier', 'console_scripts', 'towncrier')())
```

With zc.buildout which I use in the project I tested the changes the script is different and does this:
```
...
import towncrier._shell

if __name__ == '__main__':
    sys.exit(towncrier._shell.cli())
```

I haven't looked deeper yet, but it seems like the ``cli.add_command`` stuff is lost in the process somehow. I never had issues like this with a simple console script entry point like that.

**Comment by fschulze** [comment]
I would say this issue can be closed for now, because the source of the problem is most likely not towncrier but either click or zc.buildout or their interaction.

**Comment by altendky** [comment]
You have `...` instead of the rest of the code.  Is it using the wrong Python?  Maybe hand tweak the generated script to `print(towncrier._shell)` first and see what file it is referencing.  But yeah, if you pin to an old version...?  Anyways...  glad we got something identified.

**Comment by fschulze** [comment]
I checked that I get to the correct code, that is how I noticed the ``cli.add_command`` stuff is there and executed. It just behaves as if only the ``build`` command is there. I did have a pin for click 6.7 which I updated, but that didn't fix it either. There may be other pinned dependencies which I didn't check. That was when I decided to stop investigating.

Anyway, thanks for caring and if I ever feel the need to investigate further, I'll let you know my findings.
